### PR TITLE
New completion service

### PR DESCRIPTION
**Pull request by 333fred**
Use the new completion service being added in https://github.com/OmniSharp/omnisharp-roslyn/pull/1877. This enables things like override completion, as well as overall faster completion due to less filtering and preprocessing being done on the server side.

Fixes https://github.com/OmniSharp/omnisharp-vscode/issues/1044.
Fixes https://github.com/OmniSharp/omnisharp-vscode/issues/3687.

### CODE REVIEW

**Comment by 333fred** [comment]
I added a new test for override completion that should hopefully fail until the roslyn dependency is updated. If it _doesn't_ fail, I'll be concerned...

**Comment by bobaoapae** [comment]
Finally, Waiting for merge and new release

**Comment by 333fred** [comment]
@JoeRobich this now has an option for import completion as well.

File: package.json [hunk 1]
```diff
@@ -777,6 +777,11 @@
           "scope": "machine",
           "description": "Enables support for decompiling external references instead of viewing metadata."
         },
+        "omnisharp.enableImportCompletion": {
+          "type": "boolean",
+          "default": false,
+          "description": "Enables support for showing unimported types and unimported extension methods in completion lists. When committed, the appropriate using statement will be added at the top of the current file. This can have a negative impact on initial completion responsiveness, particularly for the first few completion sessions after opening a solution."
```
Comment (JoeRobich) [inline package.json:783]:
```suggestion
          "description": "Enables support for showing unimported types and unimported extension methods in completion lists. When committed, the appropriate using directive will be added at the top of the current file. This option can have a negative impact on initial completion responsiveness, particularly for the first few completion sessions after opening a solution."
```

File: src/features/completionProvider.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import { CompletionItemProvider, TextDocument, Position, CompletionContext, CompletionList, CompletionItem, MarkdownString, TextEdit, Range, SnippetString } from "vscode";
+import AbstractProvider from "./abstractProvider";
+import * as protocol from "../omnisharp/protocol";
+import * as serverUtils from '../omnisharp/utils';
+import { CancellationToken, CompletionTriggerKind as LspCompletionTriggerKind, InsertTextFormat } from "vscode-languageserver-protocol";
+import { createRequest } from "../omnisharp/typeConversion";
+
+export default class OmnisharpCompletionProvider extends AbstractProvider implements CompletionItemProvider {
+
+    #lastCompletions?: Map<CompletionItem, protocol.OmnisharpCompletionItem>;
```
Comment (JoeRobich) [inline src/features/completionProvider.ts:15]: Oh snap. Is this our first private field?

**Comment by filipw** [comment]
thanks for adding the VS Code option for `enableImportCompletion` too, that will make it more discoverable

File: src/features/completionProvider.ts (hunk 2 above)
Comment (333fred) [reply src/features/completionProvider.ts:15]: I have no idea, I search "typescript private field" and found this :smile:.

**Comment by JoeRobich** [comment]
Passing CI - https://travis-ci.org/github/OmniSharp/omnisharp-vscode/builds/719717539

**Pull request merged**